### ISSUE TITLE
tf.io.write_file not working in tf.function decorated function

### ISSUE THREAD

**Issue by junghau** [tensorflow/tensorflow#30248]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04.2 and Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0-beta1
- Python version: 3.7

**Describe the current behavior**
`tf.io.write_file` creates file in eager execution but produces no output file when decorated with `@tf.function`.

**Describe the expected behavior**
`tf.io.write_file` should create an output file whether or not being decorated with `@tf.function`.

**Code to reproduce the issue**
```python
import tensorflow as tf

@tf.function
def writeJPEG_graph(img_decoded, filename):
    out = tf.cast(img_decoded, tf.uint8)
    out = tf.image.encode_jpeg(out, quality=100)
    tf.io.write_file(filename, out)
    
def writeJPEG_eager(img_decoded, filename):
    out = tf.cast(img_decoded, tf.uint8)
    out = tf.image.encode_jpeg(out, quality=100)
    tf.io.write_file(filename, out)

img = tf.fill([256,256,3], 127) # example gray image
writeJPEG_graph(img, "./tfwrite_graph.jpg") # "tfwrite_graph.jpg" not created
writeJPEG_eager(img, "./tfwrite_eager.jpg") # "tfwrite_eager.jpg" created
```

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.0 beta1 and was able to reproduce the issue.Thanks!

**Comment by noodlez04** [comment]
This also happens in 1.14, and sadly inhibits visualizing images augmented in `tf.data.Dataset` pipelines (since they are run as a Graph)

**Comment by mdanatg** [comment]
Unfortunately the fix didn't make it into the 1.14 release, and will only be available in 2.0.

In 1.14, `write_file` should still be usable using the old-style `tf.control_dependencies`.
